amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseSerializer.java

```diff
@@ -85,6 +87,10 @@
             result.put(PENDING_CHECKPOINT_STATE_KEY, DynamoUtils.createAttributeValue(lease.checkpoint().subSequenceNumber()));
         }
 
+        if(lease.hashKeyRangeForLease() != null) {
```

Comment:
How does the existing Leases in LeaseTable will have the hash ranges updated for the lease ? Are we updating all the fields in Lease on every checkpoint ?

Reply:
The existing leases will have them updated as part of periodic shard syncer, when it finds missing hashrange fields during hashrange completion validation.